src/platform/datapath_linux.c

```diff
@@ -2171,6 +2171,17 @@ QuicDataPathBindingGetLocalMtu(
 #endif
 }
 
+#ifndef TEMP_FAILURE_RETRY
+#define TEMP_FAILURE_RETRY(expression)                              \
+    ({                                                              \
```

Comment:
Why does it have wrapping parentheses?

Reply:
Just force of habit, usually wrap expressions in parenthesis in macros for safety.

Reply:
They are unnecessary in this case. The curly braces already do everything.

Reply:
Actually, it fails without them.

```
../../src/platform/datapath_linux.c: In function 'QuicDataPathWorkerThread':
../../src/platform/datapath_linux.c:2177:9: error: expected expression before 'long'
 2177 |         long int FailureRetryResult = 0;                            \
      |         ^~~~
../../src/platform/datapath_linux.c:2203:13: note: in expansion of macro 'TEMP_FAILURE_RETRY'
 2203 |             TEMP_FAILURE_RETRY(
      |             ^~~~~~~~~~~~~~~~~~
```

Reply:
Ok. Weird. This code is doing something (getting a return value) from this macro that I didn't know was possible:
```
        int ReadyEventCount =
            TEMP_FAILURE_RETRY(
                epoll_wait(
                    ProcContext->EpollFd,
                    EpollEvents,
                    EpollEventCtMax,
                    -1));
```